src/Compilers/CSharp/Portable/Symbols/SynthesizedNamespaceSymbol.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using Roslyn.Utilities;
+using System.Diagnostics;
+using System;
+using Microsoft.CodeAnalysis.Emit;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols
+{
+    /// <summary>
+    /// Synthesized namespace that contains synthesized types or subnamespaces.
+    /// All its members are stored in a table on <see cref="CommonPEModuleBuilder"/>.
+    /// </summary>
+    internal sealed class SynthesizedNamespaceSymbol : NamespaceSymbol, IEquatable<SynthesizedNamespaceSymbol>
+    {
+        private readonly string _name;
+        private readonly NamespaceSymbol _containingSymbol;
+
+        public SynthesizedNamespaceSymbol(NamespaceSymbol containingNamespace, string name)
+        {
+            Debug.Assert(containingNamespace is object);
+            Debug.Assert(name is object);
+
+            _containingSymbol = containingNamespace;
+            _name = name;
+        }
+
+        public override int GetHashCode()
+            => Hash.Combine(_containingSymbol.GetHashCode(), _name.GetHashCode());
+
+        public override bool Equals(object obj)
+            => obj is SynthesizedNamespaceSymbol other && Equals(other);
+
+        public bool Equals(SynthesizedNamespaceSymbol other)
+        {
+            if (ReferenceEquals(this, other))
+            {
+                return true;
+            }
+
+            return other is object && _name.Equals(other._name) && _containingSymbol.Equals(other._containingSymbol);
+        }
+
+        internal override NamespaceExtent Extent
+            => _containingSymbol.Extent;
+
+        public override string Name
+            => _name;
+
+        public override Symbol ContainingSymbol
+            => _containingSymbol;
+
+        public override AssemblySymbol ContainingAssembly
+            => _containingSymbol.ContainingAssembly;
+
+        public override ImmutableArray<Location> Locations
+            => ImmutableArray<Location>.Empty;
+
+        public override ImmutableArray<SyntaxReference> DeclaringSyntaxReferences
+            => ImmutableArray<SyntaxReference>.Empty;
+
+        public override ImmutableArray<NamedTypeSymbol> GetTypeMembers()
+            => ImmutableArray<NamedTypeSymbol>.Empty;
+
+        public override ImmutableArray<NamedTypeSymbol> GetTypeMembers(string name)
+            => ImmutableArray<NamedTypeSymbol>.Empty;
+
+        public override ImmutableArray<NamedTypeSymbol> GetTypeMembers(string name, int arity)
+            => ImmutableArray<NamedTypeSymbol>.Empty;
+
+        public override ImmutableArray<Symbol> GetMembers()
+            => ImmutableArray<Symbol>.Empty;
```

Comment:
>ImmutableArray<Symbol>.Empty [](start = 15, length = 28)

This seems incorrect in general.

If there are no callers of the `Get*Members(...)` methods, the methods should probably `throw ExceptionUtilities.Unreachable;` instead.

Reply:
Why incorrect? The synthesized members are associated with the namespace via a look-aside table. The namespace does not have any declared members.

---
In reply to: [302277474](https://github.com/dotnet/roslyn/pull/36854#discussion_r302277474) [](ancestors = 302277474)

Reply:
A caller will get a misleading result since these namespaces each have members. Better to fail fast.

---
In reply to: [302289122](https://github.com/dotnet/roslyn/pull/36854#discussion_r302289122) [](ancestors = 302289122,302277474)

Reply:
I don't think it's misleading. Type symbols work like this as well. GetMembers on a type won't return synthesized nested types.
I *think* this might actually be called when enumerating top-level types to emit.

---
In reply to: [302290714](https://github.com/dotnet/roslyn/pull/36854#discussion_r302290714) [](ancestors = 302290714,302289122,302277474)